pkg/kubectl/cmd/util/helpers.go

```diff
@@ -38,19 +42,67 @@ import (
 	"github.com/spf13/pflag"
 )
 
+type debugError interface {
+	DebugError() (msg string, args []interface{})
+}
+
 func CheckErr(err error) {
 	if err != nil {
-		if errors.IsStatusError(err) {
-			glog.FatalDepth(1, fmt.Sprintf("Error received from API: %s", err.Error()))
+		if debugErr, ok := err.(debugError); ok {
+			glog.V(4).Infof(debugErr.DebugError())
 		}
-		if errors.IsUnexpectedObjectError(err) {
-			glog.FatalDepth(1, fmt.Sprintf("Unexpected object received from server: %s", err.Error()))
+		_, isStatus := err.(client.APIStatus)
+		switch {
+		case clientcmd.IsConfigurationInvalid(err):
+			fatal(MultilineError("Error in configuration: ", err))
+		case isStatus:
+			fatal(fmt.Sprintf("Error from server: %s", err.Error()))
+		case errors.IsUnexpectedObjectError(err):
+			fatal(fmt.Sprintf("Server returned an unexpected response: %s", err.Error()))
 		}
-		if client.IsUnexpectedStatusError(err) {
-			glog.FatalDepth(1, fmt.Sprintf("Unexpected status received from server: %s", err.Error()))
+		switch t := err.(type) {
+		case *url.Error:
+			glog.V(4).Infof("Connection error: %s %s: %v", t.Op, t.URL, t.Err)
+			switch {
```

Comment:
I had the original two cases and removed them.  I can reduce to if, although I expect cases to be added here for other conditions eventually.


> On Apr 3, 2015, at 12:50 PM, Jeff Lowdermilk <notifications@github.com> wrote:
> 
> In pkg/kubectl/cmd/util/helpers.go:
> 
> >  		}
> > -		if client.IsUnexpectedStatusError(err) {
> > -			glog.FatalDepth(1, fmt.Sprintf("Unexpected status received from server: %s", err.Error()))
> > +		switch t := err.(type) {
> > +		case *url.Error:
> > +			glog.V(4).Infof("Connection error: %s %s: %v", t.Op, t.URL, t.Err)
> > +			switch {
> Any reason for single case switch over if?
> 
> —
> Reply to this email directly or view it on GitHub.
>